### ISSUE TITLE
Initial build fails with column size errors

### ISSUE THREAD

**Issue by mihirshah-sg** [dbt-labs/dbt-project-evaluator#463]
### Describe the bug
When running `dbt build --select package:dbt_project_evaluator`, I get a few errors when the package tries to run staging models related to column size.

### Steps to reproduce
This is currently what is in my dbt_project.yml:

```
  dbt_project_evaluator:
    +enabled: "{{ env_var('ENABLE_DBT_PROJECT_EVALUATOR', 'true') | lower == 'true' | as_bool }}"
    +tags: "Analytics Engineering"
    +meta:
      +owner: "Analytics Engineering"
    marts:
      core:
        enabled: true
      dag:
        enabled: false
        fct_direct_join_to_source:
          +enabled: false
        fct_model_fanout:
          +enabled: false
        fct_multiple_sources_joined:
          +enabled: false
        fct_source_fanout:
          +enabled: false
        fct_staging_dependent_on_staging:
          +enabled: false
      documentation:
        fct_documentation_coverage:
          +enabled: true
        fct_undocumented_models:
          +enabled: true
      performance:
        enabled: false
      structure:
        enabled: false
        fct_model_directories:
          +enabled: false
        fct_model_naming_conventions:
          +enabled: false
      tests:
        enabled: false
        fct_missing_primary_key_tests:
          +enabled: false
        fct_test_coverage:
          +enabled: false
      dispatch:
        - macro_namespace: dbt
          search_order: ['dbt_project_evaluator', 'dbt']
```
I also have a few lines disabling the package's seed file (to use my own) and enabling tests.

### Expected results
All models run.

### Actual results
Models do not run.

### Screenshots and log output
<img width="820" alt="Screenshot 2024-06-11 at 12 53 05 PM" src="https://github.com/dbt-labs/dbt-project-evaluator/assets/150361947/fee21cce-d4f4-4869-a279-20adf577feaa">

### System information
**The contents of your `packages.yml` file:**
```
  - package: dbt-labs/dbt_project_evaluator
    version: 0.6.0
```
**Which database are you using dbt with?**
- [ ] postgres
- [x] redshift
- [ ] bigquery
- [ ] snowflake
- [ ] trino/starburst
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
Core:
  - installed: 1.3.1
  - latest:    1.8.2 - Update available!

  Your version of dbt-core is out of date!
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation

Plugins:
  - postgres: 1.3.7 - Update available!
  - redshift: 1.3.1 - Update available!

  At least one plugin is out of date or incompatible with dbt-core.
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation
```

### Are you interested in contributing the fix?
No

**Comment by b-per** [comment]
Thanks for the report.

I can see that you are on a pretty old version of the package though and we did quite a few changes recently on how we handle string/varchar in Redshfit (for example [like here](https://github.com/dbt-labs/dbt-project-evaluator/issues/437))

Would you be able to upgrade the package version? This will require also upgrading your dbt version most likely so I don't know if it is possible for you.

**Comment by stevenconnorg** [comment]
I'm also running into this error, and made a work-around by changing the `dbt.type_string()` calls to `dbt.type_large_string()` in models `base_node_columns` and `base_source_columns`

**Comment by b-per** [comment]
Should be closed with #475 I believe. We can reopen it otherwise.
